labstep/protocol.py

```diff
@@ -327,6 +329,77 @@ def getSteps(self):
         steps = self.last_version['protocol_steps']
         return listToClass(steps, ProtocolStep, self.__user__)
 
+    def getDataElements(self):
+        """
+        Retrieve the Data Elements of a Protocol.
+
+        Returns
+        -------
+        :class:`~labstep.metadata.Metadata`
+            An array of objects representing the Labstep Data Elements
+            on a Protocol.
+
+        Example
+        -------
+        ::
+
+            protocol = user.getProtocol(17000)
+            metadata = exp_protocol.getDataElements()
+        """
+        metadataThread = self.last_version['metadata_thread']
+        return listToClass(
+            metadataThread['metadatas'],
+            Metadata, self.__user__
+        )
+
+    def addDataElement(self, fieldType="default", fieldName=None,
+                       value=None, date=None,
+                       quantity_amount=None, quantity_unit=None,
+                       extraParams={}):
+        """
+        Add a Data Element to a Protocol.
+
+        Parameters
+        ----------
+        fieldType (str)
+            The field type. Options are: "default", "date",
+            "quantity", or "number". The "default" type is "Text".
+        fieldName (str)
+            The name of the field.
+        value (str)
+            The value accompanying the fieldName entry.
+        date (str)
+            The date and time accompanying the fieldName entry. Must be
+            in the format of "YYYY-MM-DD HH:MM".
+        quantity_amount (float)
+            The quantity.
+        quantity_unit (str)
+            The unit accompanying the quantity_amount entry.
+
+        Returns
+        -------
+        :class:`~labstep.metadata.Metadata`
+            An object representing the new Labstep Data Element.
+
+        Example
+        -------
+        ::
+
+            protocol = user.getProtocol(17000)
+            dataElement = protocol.addDataElement(fieldName="Refractive Index",
+                                               value="1.73")
+        """
+        metadataThread = self.last_version['metadata_thread']
+        filterParams = {'metadata_thread_id': metadataThread['id'],
+                        'type': fieldType,
+                        'label': fieldName,
+                        'value': value,
+                        'date': handleDate(date),
+                        'quantity_amount': quantity_amount,
```

Comment:
Does this actually work? I suspect not as these fields have been changed to `number` and `unit` in the db. 

If so let's see a test covering it!

If not, let's find and replace to `number` and `unit` everywhere (including metadata elsewhere). This naming makes more sense anyway.

Reply:
sure I'm happy to do it but it falls outside of the core of this PR – it's clearly related but will increase scope